src/commands/topics.test.ts

```diff
@@ -918,34 +920,54 @@ describe("commands/topics.ts produceMessage()", function () {
   });
 });
 
-describe("commands/topics.ts queryTopicWithFlink()", function () {
+describe.only("commands/topics.ts queryTopicWithFlink()", function () {
   let sandbox: sinon.SinonSandbox;
   let openTextDocumentStub: sinon.SinonStub;
   let showTextDocumentStub: sinon.SinonStub;
   let getTopicViewProviderStub: sinon.SinonStub;
+  let setUriMetadataStub: sinon.SinonStub;
+  let stubResourceManager: sinon.SinonStubbedInstance<ResourceManager>;
+  let ccloudLoaderStub: sinon.SinonStubbedInstance<CCloudResourceLoader>;
 
   beforeEach(function () {
     sandbox = sinon.createSandbox();
 
     openTextDocumentStub = sandbox.stub(vscode.workspace, "openTextDocument");
     showTextDocumentStub = sandbox.stub(vscode.window, "showTextDocument");
     getTopicViewProviderStub = sandbox.stub(topicViewProvider, "getTopicViewProvider");
+
+    stubResourceManager = sandbox.createStubInstance(ResourceManager);
+    sandbox.stub(ResourceManager, "getInstance").returns(stubResourceManager);
+    setUriMetadataStub = stubResourceManager.setUriMetadata;
+
+    // Create stub for CCloudResourceLoader
+    ccloudLoaderStub = sandbox.createStubInstance(CCloudResourceLoader);
+    sandbox.stub(CCloudResourceLoader, "getInstance").returns(ccloudLoaderStub);
   });
 
   afterEach(function () {
     sandbox.restore();
   });
 
   it("should create a new document with Flink SQL language and correct placeholder query for CCloud topic", async function () {
+    const mockFlinkComputePool = { id: "test-flink-pool" };
+    const mockEnvironment = {
```

Comment:
Respell to be something like
```suggestion
    const mockEnvironment = new CCloudEnvironment({...TEST_CCLOUD_ENVIRONMENT, flinkComputePools});
```

and you should be all set and no type error. Issue is that CCLoudEnvironment is a class, not an interface, so TS really wants the assignment RHS to be definitely an instance of the class.